net/quic/platform/impl/quic_chromium_clock.cc

```diff
@@ -19,16 +25,32 @@ QuicChromiumClock::QuicChromiumClock() = default;
 
 QuicChromiumClock::~QuicChromiumClock() = default;
 
+#if defined(STARBOARD)
+void QuicChromiumClock::ZeroApproximateNow() {
+  s_approximate_now = QuicTime::Zero();
+};
+#endif
+
 QuicTime QuicChromiumClock::ApproximateNow() const {
   // At the moment, Chrome does not have a distinct notion of ApproximateNow().
```

Comment:
Note that this 8 year old "// At the moment" comment is still there in Chromium trunk, it hasn't been implemented. One of the standalone, clean things that we could and should try and land upstream. Obviously with good test coverage and stuff.

Reply:
Sure we should try, but I wouldn't be surprised to encounter some hesitation or pushback. Upstream may want a different 'staleness' heuristic than the one here that fully relies on explicit calls to `QuicChromiumClock::ZeroApproximateNow()` to trigger re-acquisition of the current time. Our devices benefit performance-wise from getting the system clock at most once per batch of packets (less often than once per packet). They may want to also re-acquire the clock after other counts or timeouts. Upstream may also not like the idea of introducing a static.

The core question why this probably wasn't implemented before is; how does one, reliably but without looking at the clock, know that enough time has passed that the approximate time is no longer accurate? Perhaps it can be implemented at a platform library level, by using fast 'elapsed time' approximations (perhaps the platform has instruction count registers), to throttle the more expensive system calls.